pandas/core/arrays/categorical.py

```diff
@@ -1686,13 +1687,20 @@ def __array__(
         >>> np.asarray(cat)
         array(['a', 'b'], dtype=object)
         """
+        if copy is False:
```

Comment:
If `lib.is_range_indexer(self._codes)` i.e. the `self.categories._values` are all unique then, a copy could be avoided?

Reply:
Probably, but I left it out, it doesn't seem vital to try and improve it.  Also fixed things up, because `take_nd` should presumably always return a copy.